clang/include/clang/Basic/AttrDocs.td

```diff
@@ -8057,3 +8057,70 @@ requirement:
   }
   }];
 }
+
+def DocCatNonBlockingNonAllocating : DocumentationCategory<"Performance Constraint Attributes"> {
+  let Content = [{
```

Comment:
It may be good to mention what the flag is that you use to enable these warnings.

Reply:
Done in the next commit(s) I push